lost/src/main/java/com/mapzen/android/lost/internal/LostClientManager.java

```diff
@@ -0,0 +1,252 @@
+package com.mapzen.android.lost.internal;
+
+import com.mapzen.android.lost.api.LocationAvailability;
+import com.mapzen.android.lost.api.LocationCallback;
+import com.mapzen.android.lost.api.LocationListener;
+import com.mapzen.android.lost.api.LocationRequest;
+import com.mapzen.android.lost.api.LocationResult;
+import com.mapzen.android.lost.api.LostApiClient;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import static com.mapzen.android.lost.api.FusedLocationProviderApi.KEY_LOCATION_CHANGED;
+
+/**
+ * Used by {@link LostApiClientImpl} to manage connected clients and by
+ * {@link FusedLocationProviderServiceImpl} to manage client's {@link LocationListener}s,
+ * {@link PendingIntent}s, and {@link LocationCallback}s.
+ */
+public class LostClientManager implements ClientManager {
+  private static final String TAG = ClientManager.class.getSimpleName();
+
+  private static LostClientManager instance;
+  private HashSet<LostApiClient> clients;
+
+  private Map<LostApiClient, Map<LocationListener, LocationRequest>> listenerToRequest;
```

Comment:
Nice catch, I safely removed the `LocationRequest` from all the data structures and repushed!